src/INode.java

```diff
@@ -348,27 +364,30 @@ private ASTValue evalQueueAssignPropertyList(WordsEnvironment environment) {
 		throw new AssertionError("Not yet implemented");
 	}
 
-	private ASTValue evalQueueMove(WordsEnvironment environment) {
+	private ASTValue evalQueueMove(WordsEnvironment environment) throws WordsException {
 		ASTValue referenceObject = children.get(0).eval(environment);
 		ASTValue identifier = children.get(1).eval(environment);
 		ASTValue direction = children.get(2).eval(environment);
-		ASTValue distance = children.get(3) != null ? children.get(3).eval(environment) : new ASTValue(1);		// Default distance is 1
+		AST distance = children.get(3);
 		ASTValue doNow = children.get(4) != null ? children.get(4).eval(environment) : null;
 		
-		// TODO: Should not evaluate distance here but instead put it on the WordsMove object
-		
 		WordsObject object;
 		if (referenceObject.type.equals(ValueType.OBJ)){
 			WordsProperty property = referenceObject.objValue.getProperty(identifier.stringValue);
-			// TODO: Check that property is in fact an object, or throw an exception
+			if (property.type != WordsProperty.PropertyType.OBJECT) {
+				throw new InvalidTypeException(this.lineNo, ValueType.OBJ.toString(), property.type.toString());
+			}
 			object = property.objProperty;
 		} else {
 			object = environment.getObject(identifier.stringValue);
 		}
 		
-		//TODO: If distance is not a num, throw an exception
+		if (direction.type != ValueType.DIRECTION) {
```

Comment:
I don't think this is necessary.  The grammar enforces that direction is a direction.  And it can never be null.  So there's nothing to check.  I suppose it doesn't hurt to have this, but perhaps a comment about it would be helpful for clarity.  An assertion would probably be even better, since it's an invariant we believe to be true and the user should never see.